setup.py

```diff
@@ -5,42 +5,40 @@
 from setuptools import setup, find_packages, Extension
 from distutils.errors import CCompilerError, DistutilsExecError, DistutilsPlatformError
 
-# change behaviour of the setup.py on readthedocs.io
+# change behaviour of the setup.py on readthedocs.org
 # https://read-the-docs.readthedocs.io/en/latest/faq.html#how-do-i-change-behavior-for-read-the-docs
 on_rtd = os.environ.get('READTHEDOCS', None) == 'True'
-if not on_rtd:
-    import numpy as np
-    install_requires=[
+
+if on_rtd:  # if we're building on readthedocs.org:
```

Comment:
Is this needed? RTD does run examples, so it needs all these modules anyway, so why exclude them from requirements here?

Reply:
I don't remember why this is the way it is. I can try removing this and see what happens.

Reply:
Huh, seems to build just fine without the RTD difference. Maybe we can remove the RTD-checking from setup.py completely.

Reply:
This code was introduced in PR #69. Apparently readthedocs has some issues with c-dependent libraries.
https://docs.readthedocs.io/en/latest/faq.html#i-get-import-errors-on-libraries-that-depend-on-c-modules

They say: 
>With Sphinx you can use the built-in autodoc_mock_imports for mocking. If such libraries are installed via setup.py, you also will need to remove all the C-dependent libraries from your install_requires in the RTD environment.

So, what we were doing seems to be correct given their advice. I'm unsure why it _doesn't_ cause a problem to include numpy in the `install_requires`, since their docs imply that it should. 

But, the docs seem to build fine anyway. 

PS: Back in 2016, it seems that I tried to do this same thing, and ended up switching back to the way things were. No idea why! :) https://github.com/PyAbel/PyAbel/commit/107170810f0f86aeec21285fc5f6bb116bdd53ad

Reply:
If RTD does not like C, then we should disable only Cython extensions (which should happen automatically, unless RTD somehow has Cython preinstalled by default, but even then there is a check `if _cython_installed and not on_rtd` later...), but all other dependences _are_ needed on RTD. So I would say that this can and better be removed.

Reply:
Sorry, should have clarified: Numpy uses C libraries and is considered a "C-dependent library" in this context.